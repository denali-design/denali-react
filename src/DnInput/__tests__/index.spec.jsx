/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { render } from 'enzyme';
import {
    DnInput,
    DnInputSizes,
    DnInputTypes,
    DnInputGroup,
    DnButton,
    DnButtonStyles,
    DnIcon,
    DnIconColors,
    DnIconSizes,
    DnIconShades,
} from '../../index';

describe('DnInput Tests', () => {
    it('should export DnInputsSizes', () => {
        expect.assertions(1);

        expect(DnInputSizes).toStrictEqual({
            default: 'default',
            medium: 'medium',
            small: 'small',
        });
    });

    it('should export DnInputTypes', () => {
        expect.assertions(1);

        expect(DnInputTypes).toStrictEqual({
            text: 'text',
            email: 'email',
            number: 'number',
            password: 'password',
        });
    });

    it('should render with default props', () => {
        expect.assertions(1);

        const rendered = render(<DnInput />);

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <div className="Foo">
                <DnInput
                    id="someId"
                    size={DnInputSizes.default}
                    type={DnInputTypes.text}
                    isActive
                    isDisabled
                    isError
                    isInverse
                    errorMessage="Error"
                    placeholderText="Just an input field"
                    name="myInput"
                />
            </div>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with Icon at front', () => {
        expect.assertions(1);

        const rendered = render(
            <DnInput
                frontIconComponent={
                    <DnIcon icon="filter-horizontal" color={DnIconColors.brand} shade={DnIconShades.shade300} />
                }
            />,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with Icon at front with error and placeholder', () => {
        expect.assertions(1);

        const rendered = render(
            <DnInput
                isError
                errorMessage="Error"
                placeholderText="Just an input field"
                frontIconComponent={
                    <DnIcon icon="filter-horizontal" color={DnIconColors.brand} shade={DnIconShades.shade300} />
                }
            />,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with Icon at back', () => {
        expect.assertions(1);

        const rendered = render(
            <DnInput backIconComponent={<DnIcon icon="close-circle-solid" color={DnIconColors.grey} />} />,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with Icon at back with error and placeholder', () => {
        expect.assertions(1);

        const rendered = render(
            <DnInput
                isError
                errorMessage="Error"
                placeholderText="Just an input field"
                backIconComponent={<DnIcon icon="close-circle-solid" color={DnIconColors.grey} />}
            />,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with button on field', () => {
        expect.assertions(1);

        const rendered = render(
            <DnInputGroup hasButton>
                <DnInput />
                <DnButton style={DnButtonStyles.solid} className="has-icon">
                    <DnIcon size={DnIconSizes.small} icon="add" />
                </DnButton>
            </DnInputGroup>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with button on field with error and placeholder', () => {
        expect.assertions(1);

        const rendered = render(
            <DnInputGroup hasButton>
                <DnInput isError errorMessage="Error" placeholderText="Just an input field" />
                <DnButton style={DnButtonStyles.solid} className="has-icon">
                    <DnIcon size={DnIconSizes.small} icon="add" />
                </DnButton>
            </DnInputGroup>,
        );

        expect(rendered).toMatchSnapshot();
    });
});
