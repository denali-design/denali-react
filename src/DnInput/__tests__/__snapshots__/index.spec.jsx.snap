// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnInput Tests should render correctly with Icon at back 1`] = `
<div
  class="input is-default has-icon-back"
>
  <input
    type="text"
  />
  <i
    class="d-icon d-close-circle-solid is-default is-grey-500"
  />
</div>
`;

exports[`DnInput Tests should render correctly with Icon at back with error and placeholder 1`] = `
<div
  class="input is-default is-error has-icon-back"
>
  <input
    placeholder="Just an input field"
    type="text"
  />
  <i
    class="d-icon d-close-circle-solid is-default is-grey-500"
  />
  <p
    class="message"
  >
    Error
  </p>
</div>
`;

exports[`DnInput Tests should render correctly with Icon at front 1`] = `
<div
  class="input is-default has-icon-front"
>
  <i
    class="d-icon d-filter-horizontal is-default is-brand-300"
  />
  <input
    type="text"
  />
</div>
`;

exports[`DnInput Tests should render correctly with Icon at front with error and placeholder 1`] = `
<div
  class="input is-default is-error has-icon-front"
>
  <i
    class="d-icon d-filter-horizontal is-default is-brand-300"
  />
  <input
    placeholder="Just an input field"
    type="text"
  />
  <p
    class="message"
  >
    Error
  </p>
</div>
`;

exports[`DnInput Tests should render correctly with button on field 1`] = `
<div
  class="input-group has-button"
>
  <span>
    <div
      class="input is-default"
    >
      <input
        type="text"
      />
    </div>
  </span>
  <span>
    <button
      class="button is-solid is-default has-icon"
      type="button"
    >
      <i
        class="d-icon d-add is-small"
      />
    </button>
  </span>
</div>
`;

exports[`DnInput Tests should render correctly with button on field with error and placeholder 1`] = `
<div
  class="input-group has-button"
>
  <span>
    <div
      class="input is-default is-error"
    >
      <input
        placeholder="Just an input field"
        type="text"
      />
      <p
        class="message"
      >
        Error
      </p>
    </div>
  </span>
  <span>
    <button
      class="button is-solid is-default has-icon"
      type="button"
    >
      <i
        class="d-icon d-add is-small"
      />
    </button>
  </span>
</div>
`;

exports[`DnInput Tests should render correctly with custom props 1`] = `
<div
  class="Foo"
>
  <div
    class="input is-active is-default is-inverse is-disabled is-error"
  >
    <input
      disabled=""
      id="someId"
      name="myInput"
      placeholder="Just an input field"
      type="text"
    />
    <p
      class="message"
    >
      Error
    </p>
  </div>
</div>
`;

exports[`DnInput Tests should render with default props 1`] = `
<div
  class="input is-default"
>
  <input
    type="text"
  />
</div>
`;
